src/xml-properties-mixin.js

```diff
@@ -0,0 +1,151 @@
+const re = require('block-re');
+const cheerio = require('cheerio');
+
+const makeNSTagSelector = require('./make-ns-tag-selector');
+
+const $ = cheerio.load(`<?xml version="1.0" ?>`, { xmlMode: true });
+
+//
+// mixin to persist data in an xml properties element, similar to how properties
+// are stored in pPr/rPr elements in ooxml.
+//
+// numbers and strings are stored as the 'val' attribute of an element, e.g.
+// properties.foo = 5 ==> <foo val="5"/>
+//
+// key: value objects
+// properties.foo = { bar: 5, baz: 'bat' } ==> <foo bar="5", baz="bat"/>
+//
+// array elements
+// properties.push('foo', {bar: 5})
+// properties.push('foo', {bar: 6})
+// ==> <foo bar="5"/><foo bar="6"/>
+//
+
+let XmlPropertiesMixin = (superclass) => class extends superclass {
+  constructor(root, propertiesTag, namespace) {
+    if (!propertiesTag) {
+      throw new Error(`need propertiesTag for XmlProperties`);
+    }
+    super();
+      // accept either an XmlProperties object (with a .$root property) or a dom element
```

Comment:
Although having thought about it a lot since yesterday, I think that it's valuable and efficient for an object that wraps another object to be created from either a wrapper or a wrapee (similar to how you can $(wrap) either a dom element or a jQuery object and get back a jQuery object). I think that this pattern eliminates a lot of debug cycles, especially when the codebase is immature and iterating rapidly. I'd like to figure out how to take advantage of this pattern without running into your polymorphism vendetta.